ext/scripts/package_mgmt_utils.pm

```diff
@@ -13,6 +13,31 @@ sub generate_joined_and_transformed_string_from_file{
     return $final_string;
 }
 
+
+sub generate_joined_and_transformed_string_from_files {
+    my ($element_separator, $combining_template, $templates_ref, $separators_ref, $rendered_line_transformation_functions_ref, $files_ref) = @_;
+    my %transformed_lines;
+    foreach my $file ( @$files_ref ) {
+        my @transformed_lines_for_current_file = generate_transformed_lines_for_templates(
+            $file, $element_separator, $templates_ref, $rendered_line_transformation_functions_ref);
+        foreach my $lines_ref ( @transformed_lines_for_current_file ) {
+            foreach my $line ( @$lines_ref) {
+                $transformed_lines{$line} = 1;
```

Comment:
this is a map used as set, I guess, maybe call `transformed_lines` then `transformed_lines_set`

Reply:
Changed the code according to our discussion on the phone.